src/builtins/exec.rs

```diff
@@ -0,0 +1,47 @@
+use crate::builtins::InternalCommandContext;
+use crate::process::ExitStatus;
+use nix::errno::Errno;
+use std::ffi::CString;
+use std::io::Write;
+
+pub fn command(ctx: &mut InternalCommandContext) -> ExitStatus {
+    if let Some(command) = ctx.argv.get(1) {
+        let command = match CString::new(command.as_bytes()) {
+            Ok(args) => args,
+            Err(_) => {
+                writeln!(ctx.stderr, "nsh: exec: nul found in inputs").ok();
+                return ExitStatus::Return;
+            }
+        };
+        // args should include `command`
+        let args = match ctx.argv[1..]
+            .into_iter()
+            .map(|s| CString::new(s.as_bytes()))
+            .collect::<Result<Vec<_>, _>>()
+        {
+            Ok(args) => args,
+            Err(_) => {
+                writeln!(ctx.stderr, "nsh: exec: nul found in inputs").ok();
+                return ExitStatus::Return;
+            }
+        };
+
+        match nix::unistd::execvp(&command, &args) {
+            Ok(never) => match never {},
```

Comment:
Nit: I believe we can assume execvp won't return a success value.

```suggestion
            Ok(_) => unreachable!(),
```

Reply:
Exactly. Because so, I'm matching against `Infallible`, which is stronger assertion than ignoring value and calling `unreachable` .

Reply:
Sounds better! I have never heard of `never`. Thank you for explanation 😃